### PR TITLE
19543: TGW Connect and Connect Peer

### PR DESCRIPTION
**Pull request by CyrusJavan**
Add two new resources:
- aviatrix_aws_tgw_connect
- aviatrix_aws_tgw_connect_peer

Unit Test:
Create, Read, Delete, Import aviatrix_aws_tgw_connect: PASS
Create, Read, Delete, Import aviatrix_aws_tgw_connect_peer: PASS

### CODE REVIEW

File: docs/resources/aviatrix_aws_tgw_connect.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+subcategory: "TGW Orchestrator"
+layout: "aviatrix"
+page_title: "Aviatrix: aviatrix_aws_tgw_connect"
+description: |- Creates and manages Aviatrix AWS TGW Connect connections
```
Comment (shuoliugit) [inline docs/resources/aviatrix_aws_tgw_connect.md]: I remember we change line in the description in other .md files. Is it required?

**Review by shuoliugit — APPROVED** [review]

File: aviatrix/resource_aviatrix_aws_tgw_connect.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+				ForceNew:    true,
+				Description: "Transport Attachment VPC ID.",
+			},
+			"security_domain_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Security Domain Name.",
+			},
+
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_connect.go]: Remove this empty line.

File: aviatrix/resource_aviatrix_aws_tgw_connect_peer.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.IsCIDR,
+				},
+				Required:    true,
+				ForceNew:    true,
+				Description: "Set of BGP Inside CIDR Blocks.",
+			},
+
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_connect_peer.go]: Remove this empty line.

File: aviatrix/resource_aviatrix_aws_tgw_connect_peer.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+			"tgw_gre_address": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Description:  "AWS TGW GRE IP Address.",
+				ValidateFunc: validation.IsIPAddress,
+			},
+
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_connect_peer.go]: This too.

File: aviatrix/resource_aviatrix_aws_tgw_connect_peer_test.go [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+%s
+
+resource "aviatrix_aws_tgw" "test_aws_tgw" {
+  account_name          = aviatrix_account.aws.account_name
+  aws_side_as_number    = "64512"
+  region                = "%[3]s"
+  tgw_name              = "aws-tgw-%[2]s"
+  manage_vpc_attachment = false
+  manage_transit_gateway_attachment = false
+
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_connect_peer_test.go:63]: Format: line up "=" with "manage_transit_gateway_attachment".

File: aviatrix/resource_aviatrix_aws_tgw_connect_peer_test.go [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+  security_domain_name = "Shared_Service_Domain"
+  vpc_account_name     = aviatrix_account.aws.account_name
+  vpc_id               = aviatrix_vpc.tgw_attach_vpc.vpc_id
+}
+resource "aviatrix_aws_tgw_connect" "test_aws_tgw_connect" {
+  tgw_name = aviatrix_aws_tgw.test_aws_tgw.tgw_name
+  connection_name = "aws-tgw-connect-%[2]s"
+  attachment_name = aviatrix_aws_tgw_vpc_attachment.aws_tgw_vpc_attachment.vpc_id
+  security_domain_name = aviatrix_aws_tgw_vpc_attachment.aws_tgw_vpc_attachment.security_domain_name
+}
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_connect_peer_test.go:113]: Format: line up "=" with "security_domain_name".

File: aviatrix/resource_aviatrix_aws_tgw_connect_test.go [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+	return fmt.Sprintf(`
+%s
+
+resource "aviatrix_aws_tgw" "test_aws_tgw" {
+  account_name          = aviatrix_account.aws.account_name
+  aws_side_as_number    = "64512"
+  region                = "%[3]s"
+  tgw_name              = "aws-tgw-%[2]s"
+  manage_vpc_attachment = false
+  manage_transit_gateway_attachment = false
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_connect_test.go:62]: Format: same.

File: aviatrix/resource_aviatrix_aws_tgw_connect_test.go [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+  security_domain_name = "Shared_Service_Domain"
+  vpc_account_name     = aviatrix_account.aws.account_name
+  vpc_id               = aviatrix_vpc.tgw_attach_vpc.vpc_id
+}
+resource "aviatrix_aws_tgw_connect" "test_aws_tgw_connect" {
+	tgw_name = aviatrix_aws_tgw.test_aws_tgw.tgw_name
+	connection_name = "aws-tgw-connect-%[2]s"
+	attachment_name = aviatrix_aws_tgw_vpc_attachment.aws_tgw_vpc_attachment.vpc_id
+	security_domain_name = aviatrix_aws_tgw_vpc_attachment.aws_tgw_vpc_attachment.security_domain_name
+}
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_connect_test.go:113]: Format: Intent and "="

File: goaviatrix/aws_tgw_connect.go [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+func (a *AwsTgwConnect) ID() string {
+	return fmt.Sprintf("%s~~%s", a.TgwName, a.ConnectionName)
+}
+
+type AwsTgwConnectPeer struct {
+	Action string `form:"action"`
+	CID    string `form:"CID"`
+
```
Comment (zhiqiangjin) [inline goaviatrix/aws_tgw_connect.go]: Remove this empty line.

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**